python/requirements.txt

```diff
@@ -25,7 +25,7 @@ grpcio == 1.54.2; sys_platform == "darwin"
 grpcio >= 1.54.2; sys_platform != "darwin"
 numpy>=1.20
 
-pyarrow >= 6.0.1
+pyarrow >= 6.0.1,<18
```

Comment:
do we want to just pin it for macos? is it also breaking on linux?

so `pip install ray[data] pyarrow` will break for some ray data workflow now? what is the breakage look like? do we need to communicate to user anything?

Reply:
> so pip install ray[data] pyarrow will break for some ray data workflow now?

`pip install ray[data] pyarrow` on my macbook works for me, as well as building from source from latest master. running ray data code after installation works. it installed pyarrow 18. 

> it seems that the pyarrow x86_64 wheel on osx now wants the system to have cmake now.

what are the implications of this? is that something additional we need to add as a dependency for macos specifically?

Reply:
for now, pinned in just macos, since it's passing on linux.

Reply:
> pip install ray[data] pyarrow on my macbook works for me, as well as building from source from latest master. running ray data code after installation works. it installed pyarrow 18.

it might be intel silicon only.. but I am not sure. my guess is that the intel silicon wheel is not packaged correctly for the latest release.. and requires more compiling.

> what are the implications of this? is that something additional we need to add as a dependency for macos specifically?

in the logs the `pip install` error says `cmake` not found.